torch/_export/exported_program.py

```diff
@@ -78,6 +77,8 @@ def __init__(
         graph_signature: ExportGraphSignature,
         call_spec: CallSpec,
         state_dict: Dict[str, Any],
+        symbol_to_range: Dict[sympy.Symbol, RangeConstraint],
+        equality_constraints: Dict[NodeDim, List[NodeDim]],
```

Comment:
Can you explain why you need different indexes for `symbol_to_range` and `equality_constraints` here? Is it because of inline constraints, which can only be range-based?

Basically I'd have expected / liked `equality_constraints` to be `Dict[sympy.Symbol, List[sympy.Symbol]`, so the question is, why not?

Reply:
With the [`test_runtime_assert_equality_constraint`](https://www.internalfb.com/code/fbsource/[e080abab76d4fae21f2a60313b200afd1333697e]/fbcode/caffe2/test/export/test_passes.py?lines=332) test, it traces the two input dimensions as having the same symbol `s0`, so then doing an assertion of `s0 == s0` doesn't make sense.

Alternatively, we can store nothing, and then when running this pass, go through each dimension and find all the symbols, and for each symbol which maps to more than 1 dimension, add the equality assertion. For example, if arg0.shape[0] = s0 and arg1.shape[0] = s0, then we'll add the assert(arg0.shape[0] == arg1.shape[0]). Actually maybe this makes more sense. I'm not sure if it's possible to trace 2 input dimensions as having the same shape but different symbols...right?

Reply:
Oh I see. Then what you're doing makes total sense, thanks; no need to change anything. NodeDim is basically 1-1 with sources, and of course the same symbol can have multiple sources.